### ISSUE TITLE
FLAC to AAC transcode won't play on iOS

### ISSUE THREAD

**Issue by ChickenSupreme** [leinelissen/jellyfin-audio-player#69]
**Media Info of the file**
```
$ mediainfo /media/music/10235967_Soul_In_The_Dark_Extended_Mix.flac 
General
Complete name                            : /media/music/10235967_Soul_In_The_Dark_Extended_Mix.flac
Format                                   : FLAC
Format/Info                              : Free Lossless Audio Codec
File size                                : 43.4 MiB
Duration                                 : 5 min 25 s
Overall bit rate mode                    : Variable
Overall bit rate                         : 1 119 kb/s
Album                                    : Soul In The Dark
Track name                               : Soul In The Dark (Extended Mix)
Track name/Position                      : 1
Performer                                : Adip Kiyoi, Vika
Genre                                    : Trance
Description                              : Purchased at Beatport.com
Recorded date                            : 2018-02-19
Writing application                      : Lavf58.76.100
Cover                                    : Yes / Yes
Cover type                               : Publisher logo / Cover (front)
Cover MIME                               : image/png / image/png
TBPM                                     : 136
TDOR                                     : 2018-02-19
encoded_by                               : Beatport
TFLT                                     : AIFF
TIT1                                     : Suanda Voice
TKEY                                     : Gbmin
publisher                                : Suanda Voice
TSRC                                     : GBKQU1811544

Audio
Format                                   : FLAC
Format/Info                              : Free Lossless Audio Codec
Duration                                 : 5 min 25 s
Bit rate mode                            : Variable
Bit rate                                 : 1 109 kb/s
Channel(s)                               : 2 channels
Channel layout                           : L R
Sampling rate                            : 44.1 kHz
Bit depth                                : 16 bits
Compression mode                         : Lossless
Stream size                              : 43.0 MiB (99%)
Writing library                          : Lavf58.76.100
```

**Logs**
```
[2022-01-03 14:54:49.600 +00:00] [INF] [187] Jellyfin.Api.Controllers.UniversalAudioController: GetPostedPlaybackInfo profile: DeviceProfile { Name: null, Id: null, Identification: null, FriendlyName: null, Manufacturer: null, ManufacturerUrl: null, ModelName: null, ModelDescription: null, ModelNumber: null, ModelUrl: null, SerialNumber: null, EnableAlbumArtInDidl: False, EnableSingleAlbumArtLimit: False, EnableSingleSubtitleLimit: False, SupportedMediaTypes: "Audio,Photo,Video", UserId: null, AlbumArtPn: null, MaxAlbumArtWidth: 0, MaxAlbumArtHeight: 0, MaxIconWidth: null, MaxIconHeight: null, MaxStreamingBitrate: 8000000, MaxStaticBitrate: 8000000, MusicStreamingTranscodingBitrate: 128000, MaxStaticMusicBitrate: null, SonyAggregationFlags: null, ProtocolInfo: null, TimelineOffsetSeconds: 0, RequiresPlainVideoItems: False, RequiresPlainFolders: False, EnableMSMediaReceiverRegistrar: False, IgnoreTranscodeByteRangeRequests: False, XmlRootAttributes: [], DirectPlayProfiles: [DirectPlayProfile { Container: "mp3", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "aac", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "m4a", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "m4b", AudioCodec: "aac", VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "alac", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "m4a", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "m4b", AudioCodec: "alac", VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "flac", AudioCodec: "ogg", VideoCodec: null, Type: Audio }], TranscodingProfiles: [TranscodingProfile { Container: "aac", Type: Audio, VideoCodec: null, AudioCodec: "aac", Protocol: "http", EstimateContentLength: False, EnableMpegtsM2TsMode: False, TranscodeSeekInfo: Auto, CopyTimestamps: False, Context: Streaming, EnableSubtitlesInManifest: False, MaxAudioChannels: null, MinSegments: 0, SegmentLength: 0, BreakOnNonKeyFrames: False }], ContainerProfiles: [], CodecProfiles: [], ResponseProfiles: [], SubtitleProfiles: [] }
[2022-01-03 14:54:49.601 +00:00] [INF] [187] Jellyfin.Api.Helpers.MediaInfoHelper: User policy for "<redacted>". EnableAudioPlaybackTranscoding: True
[2022-01-03 14:54:49.602 +00:00] [INF] [187] Jellyfin.Api.Helpers.MediaInfoHelper: Profile: "Unknown Profile", No audio direct play profiles found for "/media/music/10235967_Soul_In_The_Dark_Extended_Mix.flac" with codec "flac"
[2022-01-03 14:54:49.602 +00:00] [INF] [187] Jellyfin.Api.Helpers.MediaInfoHelper: Profile: "Unknown Profile", No audio direct play profiles found for "/media/music/10235967_Soul_In_The_Dark_Extended_Mix.flac" with codec "flac"
[2022-01-03 14:54:49.602 +00:00] [INF] [187] Jellyfin.Api.Helpers.MediaInfoHelper: Profile: "Unknown Profile", No audio direct play profiles found for "/media/music/10235967_Soul_In_The_Dark_Extended_Mix.flac" with codec "flac"
[2022-01-03 14:54:49.603 +00:00] [INF] [187] Jellyfin.Api.Helpers.TranscodingJobHelper: /usr/lib/jellyfin-ffmpeg/ffmpeg -i file:"/media/music/10235967_Soul_In_The_Dark_Extended_Mix.flac" -threads 0 -vn -ab 384000 -ac 2 -id3v2_version 3 -write_id3v1 1 -y "/config/transcodes/e17f749dc8e3629792e5b698fcf42a5e.aac"
[2022-01-03 14:54:49.759 +00:00] [INF] [167] Jellyfin.Api.Controllers.UniversalAudioController: GetPostedPlaybackInfo profile: DeviceProfile { Name: null, Id: null, Identification: null, FriendlyName: null, Manufacturer: null, ManufacturerUrl: null, ModelName: null, ModelDescription: null, ModelNumber: null, ModelUrl: null, SerialNumber: null, EnableAlbumArtInDidl: False, EnableSingleAlbumArtLimit: False, EnableSingleSubtitleLimit: False, SupportedMediaTypes: "Audio,Photo,Video", UserId: null, AlbumArtPn: null, MaxAlbumArtWidth: 0, MaxAlbumArtHeight: 0, MaxIconWidth: null, MaxIconHeight: null, MaxStreamingBitrate: 8000000, MaxStaticBitrate: 8000000, MusicStreamingTranscodingBitrate: 128000, MaxStaticMusicBitrate: null, SonyAggregationFlags: null, ProtocolInfo: null, TimelineOffsetSeconds: 0, RequiresPlainVideoItems: False, RequiresPlainFolders: False, EnableMSMediaReceiverRegistrar: False, IgnoreTranscodeByteRangeRequests: False, XmlRootAttributes: [], DirectPlayProfiles: [DirectPlayProfile { Container: "mp3", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "aac", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "m4a", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "m4b", AudioCodec: "aac", VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "alac", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "m4a", AudioCodec: null, VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "m4b", AudioCodec: "alac", VideoCodec: null, Type: Audio }, DirectPlayProfile { Container: "flac", AudioCodec: "ogg", VideoCodec: null, Type: Audio }], TranscodingProfiles: [TranscodingProfile { Container: "aac", Type: Audio, VideoCodec: null, AudioCodec: "aac", Protocol: "http", EstimateContentLength: False, EnableMpegtsM2TsMode: False, TranscodeSeekInfo: Auto, CopyTimestamps: False, Context: Streaming, EnableSubtitlesInManifest: False, MaxAudioChannels: null, MinSegments: 0, SegmentLength: 0, BreakOnNonKeyFrames: False }], ContainerProfiles: [], CodecProfiles: [], ResponseProfiles: [], SubtitleProfiles: [] }
[2022-01-03 14:54:49.759 +00:00] [INF] [167] Jellyfin.Api.Helpers.MediaInfoHelper: User policy for "<redacted>". EnableAudioPlaybackTranscoding: True
[2022-01-03 14:54:49.759 +00:00] [INF] [167] Jellyfin.Api.Helpers.MediaInfoHelper: Profile: "Unknown Profile", No audio direct play profiles found for "/media/music/10235967_Soul_In_The_Dark_Extended_Mix.flac" with codec "flac"
[2022-01-03 14:54:49.760 +00:00] [INF] [167] Jellyfin.Api.Helpers.MediaInfoHelper: Profile: "Unknown Profile", No audio direct play profiles found for "/media/music/10235967_Soul_In_The_Dark_Extended_Mix.flac" with codec "flac"
[2022-01-03 14:54:49.760 +00:00] [INF] [167] Jellyfin.Api.Helpers.MediaInfoHelper: Profile: "Unknown Profile", No audio direct play profiles found for "/media/music/10235967_Soul_In_The_Dark_Extended_Mix.flac" with codec "flac"
[2022-01-03 14:54:49.760 +00:00] [INF] [167] Jellyfin.Api.Helpers.TranscodingJobHelper: /usr/lib/jellyfin-ffmpeg/ffmpeg -i file:"/media/music/10235967_Soul_In_The_Dark_Extended_Mix.flac" -threads 0 -vn -ab 384000 -ac 2 -id3v2_version 3 -write_id3v1 1 -y "/config/transcodes/fcbc3a5642291c2ba370b504ded25c80.aac"
[2022-01-03 14:54:53.530 +00:00] [INF] [59] Jellyfin.Api.Helpers.TranscodingJobHelper: FFmpeg exited with code 0
[2022-01-03 14:54:53.711 +00:00] [INF] [211] Jellyfin.Api.Helpers.TranscodingJobHelper: FFmpeg exited with code 0
[2022-01-03 14:54:59.726 +00:00] [INF] [28] Jellyfin.Api.Helpers.TranscodingJobHelper: Transcoding kill timer stopped for JobId "dcfbbaab5b7c46788bbbde8a04ec5a64" PlaySessionId "f9c11d6e2f944ed184a226c90bd48932". Killing transcoding
[2022-01-03 14:54:59.726 +00:00] [INF] [28] Jellyfin.Api.Helpers.TranscodingJobHelper: Deleting partial stream file(s) "/config/transcodes/e17f749dc8e3629792e5b698fcf42a5e.aac"
```

**FFmpeg Logs**
```
/Audio/ab0256fbb4571890535d699ae53f9001/universal

{"Protocol":0,"Id":"ab0256fbb4571890535d699ae53f9001","Path":"/media/music/10235967_Soul_In_The_Dark_Extended_Mix.flac","EncoderPath":null,"EncoderProtocol":null,"Type":0,"Container":"flac","Size":45480988,"Name":"10235967_Soul_In_The_Dark_Extended_Mix","IsRemote":false,"ETag":"12a860b0955bd23041cf4ceaaa09bdcc","RunTimeTicks":3251470848,"ReadAtNativeFramerate":false,"IgnoreDts":false,"IgnoreIndex":false,"GenPtsInput":false,"SupportsTranscoding":true,"SupportsDirectStream":true,"SupportsDirectPlay":true,"IsInfiniteStream":false,"RequiresOpening":false,"OpenToken":null,"RequiresClosing":false,"LiveStreamId":null,"BufferMs":null,"RequiresLooping":false,"SupportsProbing":true,"VideoType":null,"IsoType":null,"Video3DFormat":null,"MediaStreams":[{"Codec":"flac","CodecTag":null,"Language":null,"ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/44100","CodecTimeBase":"1/44100","Title":null,"VideoRange":null,"localizedUndefined":null,"localizedDefault":null,"localizedForced":null,"DisplayTitle":"FLAC - Stereo","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":"stereo","BitRate":1119025,"BitDepth":16,"RefFrames":null,"PacketLength":null,"Channels":2,"SampleRate":44100,"IsDefault":false,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":null,"Type":0,"AspectRatio":null,"Index":0,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null},{"Codec":"png","CodecTag":null,"Language":null,"ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":"Publisher/Studio logotype","TimeBase":"1/90000","CodecTimeBase":"0/1","Title":null,"VideoRange":null,"localizedUndefined":null,"localizedDefault":null,"localizedForced":null,"DisplayTitle":null,"NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":null,"BitRate":null,"BitDepth":null,"RefFrames":1,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":false,"IsForced":false,"Height":500,"Width":500,"AverageFrameRate":null,"RealFrameRate":90000,"Profile":null,"Type":3,"AspectRatio":"1:1","Index":1,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":"rgb24","Level":-99,"IsAnamorphic":null},{"Codec":"png","CodecTag":null,"Language":null,"ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":"Cover (front)","TimeBase":"1/90000","CodecTimeBase":"0/1","Title":null,"VideoRange":null,"localizedUndefined":null,"localizedDefault":null,"localizedForced":null,"DisplayTitle":null,"NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":null,"BitRate":null,"BitDepth":null,"RefFrames":1,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":false,"IsForced":false,"Height":500,"Width":500,"AverageFrameRate":null,"RealFrameRate":90000,"Profile":null,"Type":3,"AspectRatio":"1:1","Index":2,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":"rgb24","Level":-99,"IsAnamorphic":null}],"MediaAttachments":[],"Formats":[],"Bitrate":1119025,"Timestamp":null,"RequiredHttpHeaders":{},"TranscodingUrl":null,"TranscodingSubProtocol":null,"TranscodingContainer":null,"AnalyzeDurationMs":null,"DefaultAudioStreamIndex":null,"DefaultSubtitleStreamIndex":null}

/usr/lib/jellyfin-ffmpeg/ffmpeg -i file:"/media/music/10235967_Soul_In_The_Dark_Extended_Mix.flac" -threads 0 -vn -ab 384000 -ac 2 -id3v2_version 3 -write_id3v1 1 -y "/config/transcodes/fcbc3a5642291c2ba370b504ded25c80.aac"


ffmpeg version 4.3.2-Jellyfin Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 8 (Debian 8.3.0-6)
  configuration: --prefix=/usr/lib/jellyfin-ffmpeg --target-os=linux --extra-version=Jellyfin --disable-doc --disable-ffplay --disable-shared --disable-libxcb --disable-sdl2 --disable-xlib --enable-gpl --enable-version3 --enable-static --enable-libfontconfig --enable-fontconfig --enable-gmp --enable-gnutls --enable-libass --enable-libbluray --enable-libdrm --enable-libfreetype --enable-libfribidi --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libdav1d --enable-libwebp --enable-libvpx --enable-libx264 --enable-libx265 --enable-libzvbi --enable-libzimg --arch=amd64 --enable-opencl --enable-vaapi --enable-amf --enable-libmfx --enable-vdpau --enable-cuda --enable-cuda-llvm --enable-cuvid --enable-nvenc --enable-nvdec --enable-ffnvcodec
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
Input #0, flac, from 'file:/media/music/10235967_Soul_In_The_Dark_Extended_Mix.flac':
  Metadata:
    comment         : Purchased at Beatport.com
    album           : Soul In The Dark
    TBPM            : 136
    genre           : Trance
    TDOR            : 2018-02-19
    date            : 2018-02-19
    encoded_by      : Beatport
    TFLT            : AIFF
    TIT1            : Suanda Voice
    title           : Soul In The Dark (Extended Mix)
    TKEY            : Gbmin
    artist          : Adip Kiyoi, Vika
    publisher       : Suanda Voice
    track           : 1
    TSRC            : GBKQU1811544
    encoder         : Lavf58.76.100
  Duration: 00:05:25.15, start: 0.000000, bitrate: 1119 kb/s
    Stream #0:0: Audio: flac, 44100 Hz, stereo, s16
    Stream #0:1: Video: png, rgb24(pc), 500x500 [SAR 1:1 DAR 1:1], 90k tbr, 90k tbn, 90k tbc (attached pic)
    Metadata:
      comment         : Publisher/Studio logotype
    Stream #0:2: Video: png, rgb24(pc), 500x500 [SAR 1:1 DAR 1:1], 90k tbr, 90k tbn, 90k tbc (attached pic)
    Metadata:
      comment         : Cover (front)
Stream mapping:
  Stream #0:0 -> #0:0 (flac (native) -> aac (native))
Press [q] to stop, [?] for help
Output #0, adts, to '/config/transcodes/fcbc3a5642291c2ba370b504ded25c80.aac':
  Metadata:
    comment         : Purchased at Beatport.com
    album           : Soul In The Dark
    TBPM            : 136
    genre           : Trance
    TDOR            : 2018-02-19
    date            : 2018-02-19
    encoded_by      : Beatport
    TFLT            : AIFF
    TIT1            : Suanda Voice
    title           : Soul In The Dark (Extended Mix)
    TKEY            : Gbmin
    artist          : Adip Kiyoi, Vika
    publisher       : Suanda Voice
    track           : 1
    TSRC            : GBKQU1811544
    encoder         : Lavf58.45.100
    Stream #0:0: Audio: aac (LC), 44100 Hz, stereo, fltp (16 bit), 384 kb/s
    Metadata:
      encoder         : Lavc58.91.100 aac
size=    1792kB time=00:00:41.33 bitrate= 355.2kbits/s speed=82.5x    
size=    4096kB time=00:01:26.77 bitrate= 386.7kbits/s speed=86.6x    
size=    6144kB time=00:02:12.23 bitrate= 380.6kbits/s speed=  88x    
size=    8192kB time=00:02:55.58 bitrate= 382.2kbits/s speed=87.6x    
size=   10496kB time=00:03:43.35 bitrate= 385.0kbits/s speed=89.2x    
size=   12800kB time=00:04:30.88 bitrate= 387.1kbits/s speed=90.2x    
size=   14848kB time=00:05:16.34 bitrate= 384.5kbits/s speed=90.2x    
size=   15223kB time=00:05:25.14 bitrate= 383.5kbits/s speed=83.1x    
video:0kB audio:15127kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.632852%
[aac @ 0x55810a20eb40] Qavg: 1992.720
```

**Stats for Nerds Screenshots**

**Server System (please complete the following information):**
 - OS: Docker (official jellyfin/jellyfin image) on Ubuntu Server 20.04
 - Jellyfin Version: 10.7.7
 - Hardware settings & device: software transcoding
 - Reverse proxy: nginx
 - Other hardware notes: media mounted in CIFS/SMB share
 
**Client System (please complete the following information):**
 - Device: Apple iPhone SE
 - OS: iOS
 - Client: Jellyfin Audio Player https://github.com/leinelissen/jellyfin-audio-player

When trying to play a lossless FLAC music track on iOS the playback won't start. I thought it might be a server problem because of this log message:

[2022-01-03 14:54:59.726 +00:00] [INF] [28] Jellyfin.Api.Helpers.TranscodingJobHelper: Transcoding kill timer stopped for JobId "dcfbbaab5b7c46788bbbde8a04ec5a64" PlaySessionId "f9c11d6e2f944ed184a226c90bd48932". Killing transcoding

but that seems to be a red herring. I copied the transcode (fcbc3a5642291c2ba370b504ded25c80.aac) to my desktop and played it with VLC so I don't think there's anything wrong with the transcode.

It might still be a server problem but I have no idea how to diagnose it myself so I thought I'd start by posting an issue here.

By the way, thanks so much for adding playlist support, it's very handy!

**Comment by leinelissen** [comment]
Hi there! Thanks for submitting this issue. I'll need to take a closer look later, but the following is true for iOS:
* iOS doesn't have native support for playing FLAC files
* Including a native decoder for FLAC is non-feasible due to the stack of dependencies we need for making it work with React Native
* iOS does support ALAC, but Jellyfin cannot transcode to ALAC at this point in time.
* Change is promised for the next Jellyfin major release in terms of encoding, but we'll have to see what sticks in terms of features
* Because of all of the above, we force Jellyfin to transcode FLAC to M4A AACs. 

I've developed the playback with demo FLAC files and for me locally, it appeared to work. I'll see if I have the time to do a deep dive with how everything works out on iOS. Maybe I can take a look at catching playback errors and reporting them somehow so these issues become easier to debug.

At the very least, thanks for submitting the detailed issue!

**Comment by leinelissen** [comment]
This issue should now be solved with FLAC support introduced in v1.27. Do let me know if you into any additional issues!
